### PR TITLE
Move plugin and interceptor classes out of experimental

### LINKED ISSUE #3582: Move plugin classes out of experimental package
### Context & proposal
Plugin support is in the stable stage. So we should move related classes out of the experimental package.

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] All plugins, interceptors, and other related classes are moved out of the experimental package.

### PR DESCRIPTION
**Pull request by amitkma**
### 🎯 Goal

Closes #3582
We have plugin-related classes which are stable now. so we should move them out of experimental.

### 🛠 Implementation details

Move the related classes out of the experimental package.

### 🧪 Testing

Run the sample apps, and they should work as earlier with no behavior change or crash.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
~- [ ] New code is covered by unit tests~
~- [ ] Comparison screenshots added for visual changes~
~- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)~

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] Release notes and docs clearly describe changes

### CODE REVIEW

**Comment by kanat** [comment]
> @kanat Does it interfere with your state module? I think this PR may create a conflict with your work.

It does. Someone has to suffer :)
Let's wait until this PR gets merged, then I'll rebase the `StateModule` branch.

**Review by leandroBorgesFerreira — APPROVED** [review]

**Pull request merged**
